dev_guide/routes.adoc

```diff
@@ -133,6 +133,22 @@ TLS termination and a default certificate, which may not match the requested
 host name, resulting in validation errors.
 ====
 
+[[creating-labeled-routes]]
+== Creating Routes with Labels
+
+You can create a route based on labels. A explanation how can be found at
+link:../architecture/core_concepts/routes.adoc#router-sharding
```

Comment:
Well I think this code snippets say something different.

https://github.com/openshift/origin/blob/master/pkg/cmd/infra/router/router.go#L53
https://github.com/openshift/origin/blob/master/pkg/cmd/infra/router/router.go#L92
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L2606

As far as I understand the code right you need labels for router selection.

In case the VAR ROUTE_LABELS or LabelSelector is not set, which is default, the else tree 'o.Labels = labels.Everything()' is in use.

For router sharding you NEED to label ALL Routers or the apps wil also be deployed to the not labeled router.

That's the purpose of this sentence ;-)